collect_app/src/main/java/org/odk/collect/android/application/Collect.java

```diff
@@ -353,24 +354,30 @@ public void setComponent(AppDependencyComponent applicationComponent) {
         applicationComponent.inject(this);
     }
 
-    /**
-     * Gets a unique, privacy-preserving identifier for the current form.
-     *
-     * @return md5 hash of the form title, a space, the form ID
-     */
     public static String getCurrentFormIdentifierHash() {
-        String formIdentifier = "";
         FormController formController = getInstance().getFormController();
         if (formController != null) {
             if (formController.getFormDef() != null) {
                 String formID = formController.getFormDef().getMainInstance()
                         .getRoot().getAttributeValue("", "id");
-                formIdentifier = formController.getFormTitle() + " " + formID;
+
+                String formVersion = formController.getFormDef().getMainInstance()
+                        .getRoot().getAttributeValue("", "version");
+                return getCurrentFormIdentifierHash(formID, formVersion);
             }
         }
 
-        return FileUtils.getMd5Hash(
-                new ByteArrayInputStream(formIdentifier.getBytes()));
+        return "";
+    }
+
+    /**
+     * Gets a unique, privacy-preserving identifier for the current form.
+     *
+     * @return md5 hash of the form title, a space, the form ID
+     */
```

Comment:
This comment should go with the zero-param variant of the method -- "current" here refers to the form that the `FormController` is keeping track of.

This method should probably be called `getFormIdentifierHash` or `getFormIdentifierHashFor` since it refers to any form.

I suggest you keep the two variants independent so that there doesn't need to be a database query in the "current form" case. It's not a huge deal but every bit of overhead adds up on slower devices.

Reply:
Done.